config/bootstrap.go

```diff
@@ -95,39 +94,36 @@ func BootstrapCustomResourceDefinitionFromFS(ctx context.Context, client apiexte
 		return fmt.Errorf("decoded CRD %s into incorrect type, got %T, wanted %T", gk.String(), rawCrd, &apiextensionsv1.CustomResourceDefinition{})
 	}
 
-	crd, err := client.Create(ctx, rawCrd, metav1.CreateOptions{})
-	if err != nil && !apierrors.IsAlreadyExists(err) {
-		return fmt.Errorf("could not create CRD %s: %w", gk.String(), err)
+	crd, err := client.Get(ctx, rawCrd.Name, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("Error fetching CRD %s: %w", gk.String(), err)
+	}
+
+	if crd != nil && crdhelpers.IsCRDConditionTrue(crd, apiextensionsv1.Established) {
+		return nil
 	}
 
-	watcher, err := client.Watch(ctx, metav1.ListOptions{
-		FieldSelector:   fields.OneTermEqualSelector("metadata.name", crd.Name).String(),
-		ResourceVersion: crd.ResourceVersion,
-	})
+	crd, err = client.Create(ctx, rawCrd, metav1.CreateOptions{})
 	if err != nil {
-		return fmt.Errorf("could not watch CRD %s: %w", gk.String(), err)
+		return fmt.Errorf("could not create CRD %s: %w", gk.String(), err)
 	}
-	for {
-		select {
-		case <-ctx.Done():
-			return fmt.Errorf("failed to wait for CRD %s to be established: %w", gk.String(), ctx.Err())
-		case event := <-watcher.ResultChan():
-			switch event.Type {
-			case watch.Added, watch.Bookmark:
-				continue
-			case watch.Modified:
-				updated, ok := event.Object.(*apiextensionsv1.CustomResourceDefinition)
-				if !ok {
-					continue
-				}
-				if crdhelpers.IsCRDConditionTrue(updated, apiextensionsv1.Established) {
-					return nil
-				}
-			case watch.Deleted:
-				return fmt.Errorf("CRD %s was deleted before being established", gk.String())
-			case watch.Error:
-				return fmt.Errorf("encountered error while watching CRD %s: %#v", gk.String(), event.Object)
+
+	done := make(chan struct{})
+	wait.Until(func() {
```

Comment:
I think the general logic we'd like is:

1. If the CRD already exists, poll until it's established
2. If the CRD doesn't exist, try to create it, then poll until it's established

WDYT?

Also, I'd use wait.PollImmediateInfiniteWithContext - that'll save you having to create a done channel.